app/controllers/donors_controller.rb

```diff
@@ -8,14 +11,29 @@ def index
   end
 
   def admin_index
-    if current_donor.admin?
-      redirect_to '/'
+    @donor = Donor.where(approved: false, admin: false)
+    @donors = Donor.not_admin
+    @pickup = FoodPickup.find_by(id: params[:id])
+    @approved_pickups = FoodPickup.approved
+    @unapproved_pickups = FoodPickup.unapproved
+    @negotiable_profiles = Donor.where(negotiable: true)
+  end
+
+  def negotiate
+    @negotiable_profile = Donor.find_by(id: params[:id])
+  end
+
+  def update_rate
+    negotiable_profile = Donor.find_by(id: params[:id])
+    @rate = negotiable_profile.update(
+      charge: params[:charge],
+      negotiable: false
+    )
+    redirect_to '/admin'
+    if @rate
```

Comment:
Are you using `@rate` in the view? Be sure to not make these instance variables if they are not being used outside this method.

Plust, `@rate` is holding a boolean value and the variable name doesn't connote that. But I'm not sure why you're setting the result of the update to a variable